silkworm/core/protocol/merge_rule_set.cpp

```diff
@@ -70,6 +72,28 @@ ValidationResult MergeRuleSet::validate_seal(const BlockHeader& header) {
     return header.nonce == BlockHeader::NonceType{} ? ValidationResult::kOk : ValidationResult::kInvalidNonce;
 }
 
+void MergeRuleSet::initialize(EVM& evm) {
+    const BlockHeader& header{evm.block().header};
+    if (header.difficulty != 0) {
+        pre_merge_rule_set_->initialize(evm);
+        return;
+    }
+
+    if (evm.revision() < EVMC_CANCUN) {
+        return;
+    }
+
+    // EIP-4788: Beacon block root in the EVM
+    SILKWORM_ASSERT(header.parent_beacon_block_root);
+    Transaction system_txn{{
+        .type = TransactionType::kSystem,
+        .to = kBeaconRootsAddress,
+        .data = Bytes{ByteView{*header.parent_beacon_block_root}},
+    }};
+    system_txn.from = kSystemAddress;
+    evm.execute(system_txn, kSystemCallGasLimit);
```

Comment:
Although the gas price is 0 (no cost) the sender nonce will be bumped. Is this expected?

Reply:
Nope, `EVM::execute` doesn't bump sender's nonce